core/cache/application_test.go

```diff
@@ -79,6 +81,147 @@ func (s *ApplicationSuite) TestConfigWatcherChange(c *gc.C) {
 	c.Check(testutil.ToFloat64(s.Gauges.ApplicationHashCacheMiss), gc.Equals, float64(2))
 }
 
+func (s *ApplicationSuite) status(value status.Status, when time.Time) status.StatusInfo {
+	return status.StatusInfo{
+		Status: value,
+		Since:  &when,
+	}
+}
+
+func (s *ApplicationSuite) TestStatusWhenSet(c *gc.C) {
+	model := s.NewModel(cache.ModelChange{
+		Name: "test",
+	})
+	appStatus := s.status(status.Active, time.Now())
```

Comment:
maybe make this "maintenance" or something other than "active" as that has special logic
or do both?

Reply:
While "active" has special meaning to us, inside this package it has no special meaning at all.